Task/models

adael2 opened the pull request:
#### What's this PR do?

The models for Estudio, Add the models of Estudio, Seccion, Subseccion, Pregunta, Opcion Respuesta, Respuesta.

#### Where should the reviewer start?

in express-app/models/estudio.js

#### Any background context you want to provide?

Check if is the correct way to create models


This template was adapted ~retrived~ from [Quickleft/Sprint.ly](https://quickleft.com/blog/pull-request-templates-make-code-review-easier/)

Ral15 committed:
express and electron combined, trouble with db and jquery

Ral15 committed:
app now starts with npm start, .DS_Store added to gitignore

Ral15 committed:
start script added to packagejson

Ral15 committed:
electron app with express and nedb, a user can be created, but having troubles at trying to use jquery

Ral15 committed:
.gitignore is now working

Ral15 committed:
login and create user implemented

Ral15 committed:
log in form is validated

Ral15 committed:
message if u did wrong

Ral15 committed:
camo included, basic structure of the project

Ral15 committed:
Merge pull request #1 from Ral15/task/connectCamo

Task/connect camo

Ral15 committed:
Login form from bootstrap template adapted to our form. Removed materialize files and instead added bootstrap files. Login view is now the main view, this means that the login view is the first viiew that appears when the app is opened.

Ral15 committed:
Login implemented, only missing tests....

Commit:
Create a document for GIT STYLE GUIDE

This guide stablishes and standardizes the way we will be working
with git. It explains the different types of braches, commit format,
how to issue pull requesr and raise issues.

This document is most the same that are in our linked project
jp2_online.

Commit:
Document for issue template

This template will prefil the issue description box, whenever anyone
reports and issue. This document is very similiar that we used in
the linked project jp2_online.

Commit:
Document to use when you do a pull request

This template will prefill the information box, whenever someone
starts a new pull request. This document is similiar that we use
in the linked project jp2_online.

Ral15 committed:
Dashboard view modified, now has a button that is disabled if no internet conenction is found, this could be useful later.

Commit:
Merge pull request #5 from Ral15/docs/pull-request

Document to use when you do a pull request

Commit:
Merge pull request #4 from Ral15/docs/git-style-guide

Create a document for GIT STYLE GUIDE

Commit:
Merge pull request #3 from Ral15/docs/issue-template

Document for issue template

Ral15 committed:
Api url added to urls.json, jquery is now included properly in the app.

Ral15 committed:
Fixed routes

Ral15 committed:
Merge pull request #2 from Ral15/task/login-form

Task/login form

Commit:
Test of Log in

This commit do a test of login launching the application.

Commit:
Add a how to test file

This commit add a how to test for the future test in the applications.

Commit:
Requested Changes

This commit resolves the request changes, I put the defult url.
I delete the log out button. I find a better form to use launch the
app without do a production. I create all possible test for log in.

Commit:
Add comments to test.

This commit is to add comments in the login test to explain what is happend.

Commit:
Delete How to do a test.

I delete this md because is better explain in the login test.

Commit:
Merge pull request #6 from Ral15/task/test-online

Task/test online

Commit:
Create models for app offline, Creating model section

This commit is for the creation of section model.

Commit:
Create new models

Creating Subsecciones, Pregunta, Opciones de Respuesta, Respuesta
and Estudio Models

Ral15 commented:
No puedes mergear directo a master, tiene que ser a develop

Ral15 commented on express-app/models/estudio.js:
```diff
@@ -0,0 +1,184 @@
+        super();
+
+        this.capturista = User;
+        this.familia = Familia;
```
Attributes names should be in english. Also, you need to import User model or this won't work.

Ral15 reviewed (changes requested):
Change attribute names to english.

Commit:
Separate models, add Embedded Documents

Separate all the models that were in Estudio in different documents.
Add the Embedded Documents extension.

Commit:
Requested Changes from Raul

Add the requires for Familias and User.

Commit:
Add .js to the require

Add .js in require for the models

Ral15 commented on express-app/models/estudio.js:
```diff
@@ -30,154 +33,17 @@ class Estudio extends Document {
+        this.seccion = [Seccion];
+        this.status = {
+          type: String,
+          choices : ['Aprobado', 'Rechazado', 'Borrador', 'Revisión', 'Eliminado']
```
choices should be stored in a const array.

Ral15 commented on express-app/models/pregunta.js:
```diff
@@ -0,0 +1,37 @@
+"use strict";
+
+//require document class from camo
+var Document = require('camo').Document;
```
this should be removed.

Ral15 commented on express-app/models/pregunta.js:
```diff
@@ -0,0 +1,37 @@
+        this.texto = String;
+        this.descripcion = String;
+        this.respuesta = Respuesta;
+    }
```
there should be a 'subseccion' attribute in the model.

Ral15 commented on express-app/models/pregunta.js:
```diff
@@ -0,0 +1,37 @@
+
+        this.texto = String;
+        this.descripcion = String;
+        this.respuesta = Respuesta;
```
a 'pregunta' doesn't have a 'respuesta', look at the models from the app online.

Ral15 commented on express-app/models/respuesta.js:
```diff
@@ -0,0 +1,48 @@
+        super();
+
+        this.eleccion = {
+          this.opcion = OpcionRespuesta
```
why do you need 'this.opcion'?

Ral15 commented on express-app/models/respuesta.js:
```diff
@@ -0,0 +1,48 @@
+        };
+        this.integrante = Integrante;
+        this.respuesta = String
+    }
```
you are missing pregunta and estudio attributes, please refer to the models created at app online.

Ral15 commented on express-app/models/subseccion.js:
```diff
@@ -0,0 +1,34 @@
+        super();
+
+        this.nombre = String;
+        this.questions = [Pregunta];
```
please refger to the models created at the app online.

Ral15 reviewed (changes requested):
refer to the models of the app online.

Commit:
Requested Changes from Raul

Changes of the models. delete requires, add embedded documents.

Ral15 commented on express-app/models/estudio.js:
```diff
@@ -14,30 +23,32 @@ var Familia = require('./familia.js');
         super();
 
-        this.capturista = User;
+        this.capturista = Number;
```
is this attribute the id from the User model? Don't you think we need all the User model?

Ral15 commented on express-app/models/seccion.js:
```diff
@@ -19,7 +20,9 @@ class Seccion extends EmbeddedDocument {
 
+        this.id = Number;
         this.nombre = String;
+        this.numero = Number;
```
what does this means?

Commit:
Add default to models, change name to camel case.

Add the default to atributes. change the name for the atributes.

Ral15 commented on express-app/models/pregunta.js:
```diff
@@ -12,27 +12,43 @@ const OpcionRespuesta = require('./opcion_respuesta.js');
+        };
+        this.idSubseccion = {
+          type : Number,
+          default : 1
```
not sure if this should have a default value since "Pregunta" will be created only once per "Estudio" and the value will "never change".

adael2 replied:
I let a Default in Id. If something we're wrong when app offline retrived the data from the app online we can see it. I put 0 as default to see it.

Ral15 commented on express-app/models/pregunta.js:
```diff
@@ -12,27 +12,43 @@ const OpcionRespuesta = require('./opcion_respuesta.js');
+        };
+        this.orden = {
+          type : Number,
+          default : 1
```
same for this one.

adael2 replied:
Same as before

Commit:
Let 0 to default id, add id attribute to respuesta

I let 0 to default ids if somenthing we're wrong when retrive the data
from online could see it. Also put the attribute id to respuesta when
estudio is rechazado.

Ral15 reviewed (approved):
:call_me_hand:
